homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -235,6 +244,31 @@ async def async_step_2sa(self, user_input, errors=None):
 
         return await self.async_step_user(user_input)
 
+    async def async_step_filter_storage(self, user_input):
+        """Select disks and volumes to add."""
+        if not self.saved_user_input:
+            self.saved_user_input = user_input
+
+        if not user_input.get(CONF_VOLUMES):
+            return self.async_show_form(
+                step_id="filter_storage",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(
+                            CONF_DISKS, default=self.syno_info.get(CONF_DISKS, [])
+                        ): cv.multi_select(self.syno_info.get(CONF_DISKS, [])),
+                        vol.Required(
+                            CONF_VOLUMES, default=self.syno_info.get(CONF_VOLUMES, [])
+                        ): cv.multi_select(self.syno_info.get(CONF_VOLUMES, [])),
```

Comment:
Devices should never be undefined
--> `self.syno_info[CONF_...]`

Reply:
In case of an virtual DSM running on virtual machine manager, there are no disks available (see #42523), but yes for volumes - they never should be undefined ... I will change it